utils/firestore.js

```diff
@@ -1381,8 +1381,10 @@ const submitSpecimen = async (biospecimenData, participantData, siteTubesList) =
             .where('state.uid', '==', participantUid)
             .get();
         const eligibilityUpdates = processMouthwashEligibility(participantSnapshot.docs[0].data());
```

Comment:
Looks like function `processMouthwashEligibility` return CID of "initialized" or undefined (null should be better than undefined). It can be better to directly return the value instead of wrapping the value in `updates` object, so that value checking can be more straightforward.

Reply:
Checking whether the snapshot is empty shouldn't be necessary, as the check that the participant exists already occurs earlier in

```
const participantSnapshot = await transaction.get(participantQuery);

            if(participantSnapshot.size === 0) {
                throw new Error('Participant with uid ' + participantUid + 'not found.');
            }
```

As for the return of `processMouthwashEligibility`, I mostly did this in case we need to do other updates at the same time, which was the case before this function had to be factored into its own function. I'm hoping to merge further update work in here in the future, so I'd rather leave those changes as easy to make as possible.